osbenchmark/worker_coordinator/runner.py

```diff
@@ -1044,8 +1054,11 @@ def calculate_recall(predictions, neighbors, top_k):
             id_field = params.get("id-field-name", "_id")
             candidates = []
             for hit in response_json['hits']['hits']:
-                if id_field in hit:  # Will add to candidates if field value is present
-                    candidates.append(hit[id_field])
+                field_value = _get_field_value(hit, id_field)
+                if field_value is None:  # Will add to candidates if field value is present
+                    self.logger.warning("No value found for field %s", id_field)
```

Comment:
Do we want to set this as a warning? If there are a lot of searches, this could bloat the log. But, ideally in a prod setting, this shouldnt happen, right?

Reply:
Yes, it shouldn't happen in prod, thats why i added as warning. I can change it to debug if you think it is likely based on your experience.

Reply:
makes sense. I guess warning is fine